lib/metasploit/framework/data_service/proxy/core.rb

```diff
@@ -115,6 +115,12 @@ def get_data_service
     return @current_data_service
   end
 
+  def log_error(exception, ui_message)
+    elog "#{ui_message}: #{exception.message}"
+    exception.backtrace.each { |line| elog "#{line}" }
+    raise Exception, "#{ui_message}: #{exception.message}. See log for more details."
```

Comment:
I overlooked this earlier, but when taking a finally pass over the code this catch my eye. We log the exception message, but lose the type and end up throwing a generic `Exception`. Should we be throwing a custom exception type here?